packages/angular-material/src/layouts/array-layout.renderer.ts

```diff
@@ -0,0 +1,165 @@
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { JsonFormsAngularService, JsonFormsBaseRenderer } from '@jsonforms/angular';
+import {
+  computeLabel,
+  ControlElement,
+  createDefaultValue,
+  generateDefaultUISchema,
+  isObjectArray,
+  isPlainLabel,
+  JsonFormsState,
+  mapDispatchToArrayControlProps,
+  mapStateToArrayControlProps,
+  OwnPropsOfControl,
+  OwnPropsOfRenderer,
+  Paths,
+  RankedTester,
+  rankWith,
+  setReadonly,
+  StatePropsOfArrayControl,
+  UISchemaElement,
+  unsetReadonly
+  } from '@jsonforms/core';
+  import { Subscription } from 'rxjs';
+
+  @Component({
+    selector: 'app-array-layout-renderer',
+    template: `
+      <div fxLayout="column" fxLayoutGap="16px" [fxHide]="hidden">
+        <mat-toolbar>
+          <h2>{{ this.label }}</h2>
+          <span fxFlex></span>
+          <button
+            mat-icon-button
+            [disabled]="readonly"
+            (click)="add()"
+            attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
+          >
+            <mat-icon>add</mat-icon>
+          </button>
+        </mat-toolbar>
+        <p *ngIf="noData">No data</p>
+        <div
+          *ngFor="
+            let item of props.data;
+            let idx = index;
+            trackBy: trackByFn;
+            last as last
+          "
+        >
+          <mat-card>
+            <mat-card-content>
+              <jsonforms-outlet
+                [renderProps]="getProps(idx)"
+              ></jsonforms-outlet>
+            </mat-card-content>
+            <mat-card-actions *ngIf="!readonly">
+              <button mat-button (click)="remove(idx)" attr.aria-label="Remove">
+                <mat-icon>remove_circle</mat-icon>
+                Remove
+              </button>
+            </mat-card-actions>
+          </mat-card>
+        </div>
+      </div>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush
+  })
+  export class ArrayLayoutRenderer extends JsonFormsBaseRenderer<ControlElement>
```

Comment:
why not extend the JsonFormsArrayControl?

Reply:
Good point! I changed the renderer so it extends `JsonFormsArrayControl`